core/src/main/java/feign/InvocationHandlerFactory.java

```diff
@@ -32,7 +32,7 @@
    */
   interface MethodHandler {
 
-    Object invoke(Object[] argv) throws Throwable;
+    Object invoke(Object proxy, Object[] argv) throws Throwable;
```

Comment:
ex this will at least break spring-cloud-netflix and possibly others.. that would mean a major version bump